packages/medusa/src/loaders/admin.ts

```diff
@@ -1,22 +1,43 @@
 import { AdminOptions, ConfigModule } from "@medusajs/types"
 import { Express } from "express"
+import fs from "fs"
+import path from "path"
 
 type Options = {
   app: Express
-  adminConfig: ConfigModule["admin"]
+  configModule: ConfigModule
+  rootDirectory: string
 }
 
 type IntializedOptions = Required<
   Pick<AdminOptions, "path" | "disable" | "outDir">
 > &
-  AdminOptions
+  AdminOptions & {
+    sources?: string[]
+  }
+
+export default async function adminLoader({
+  app,
+  configModule,
+  rootDirectory,
+}: Options) {
+  const { admin } = configModule
+
+  const sources: string[] = []
+
+  const projectSource = path.join(rootDirectory, "src", "admin")
```

Comment:
q: Not entirely sure myself, but will the `yarn dev` changes made on the project affect this at all in discovering these files?

Reply:
I think we are still getting the rootDirectory param, no? And we are reading from src files not dist, so we shouldn't have an issue due to missing dist files.

Reply:
yeah in theory it shouldnt change anything here, the root dir does not change 👍

Reply:
Yes, you're right!